lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -219,11 +226,16 @@ def edit_user(self, user, name, group, password, email):
             resource['group'] = dict(id=group_id)
         if name is not None:
             resource['name'] = name
-        if password is not None:
-            resource['password'] = password
         if email is not None:
             resource['email'] = email
 
+        # if there is a password param, but 'update_password' is 'on_create'
+        # then discard the password (since we're editing an existing user)
+        if password is not None and self.module.params['update_password'] == 'on_create':
+            password = None
```

Comment:
this works fine but the if/elif is slightly complicated to follow, one branch touching password the other resource['password']...
Consider something like this:
```
if self.module.params['update_password'] == 'on_create':
  password = None

if password is not None:
   resource['password'] = password
```